sdk/storage/Azure.Storage.Common/tests/PooledMemoryStreamTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using Azure.Core.Pipeline;
+using Azure.Storage.Shared;
+using NUnit.Framework;
+
+namespace Azure.Storage.Tests
```

Comment:
Can we have some "live" tests that actually try to push 4-8GB blocks E2E? So that we catch potential problems in layers on the top of PooledMemoryStream? (i.e. validations, retaining references preventing GC from collecting garbage).

Reply:
Is this not covered by additions to PartitionedUploaderTests where we pump in 2 blocks each maxint plus change in size? Or are you asking I add some checks to that test where I validate every rented array was returned?

Reply:
The PartitionedUploaderTests focus on single component that's being used in larger process of uploading blobs. It's good that we focus our coverage here.

What I'm saying is that we should have a test or two testing functionality of "jumbo blob" end to end. (I've found few surprises outside of "PooledMemoryStream" equivalent there by doing that).

Reply:
Well. Sorry. I think I confused Uploader with PooledMemoryStream. You're absolutely right! Please disregard this conversation.